go/spec/spec_test.go

```diff
@@ -44,8 +43,8 @@ func TestMemDatasetSpec(t *testing.T) {
 
 	assert.Equal("mem", spec.Protocol)
 	assert.Equal("", spec.DatabaseName)
-	assert.Equal("test", spec.DatasetName)
-	assert.True(spec.Path.IsEmpty())
+	assert.Equal("test", spec.Path.Dataset)
+	assert.True(spec.Path.Path.IsEmpty())
```

Comment:
There may be other tests you can improve by using this pattern. I'm not sure. There were surprisingly few changes in the test file actually.